File: CoreJava/SOLID-princples-app/src/ocp/refractor/FixedDeposit.java
package ocp.refractor;

public class FixedDeposit {
	private final String name;
	private final double amount;
	private final int period;
	IFestivalRate festivalrate;

	public FixedDeposit(String name, double amount, int period,
			IFestivalRate festivalrate) {
		super();
		this.name = name;
		this.amount = amount;
		this.period = period;
		this.festivalrate = festivalrate;
	}

	public String getName() {
		return name;
	}

	public double getAmount() {
		return amount;
	}

	public int getPeriod() {
		return period;
	}

	public double calcInterest() {
		float rate = 0;
		return amount * period * festivalrate.getRate() / 100;
	}
}

File: DotNET/Solid Principles/OCPRefractorApp/OCPRefractorApp/FixedDeposit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OCPRefractorApp
{
    class FixedDeposit
    {
        private String _name;
        private double _amount;
        private int _period;
        private IFestivalRate _festivalrate;

        public FixedDeposit(String name, double amount, int period, IFestivalRate festivalrate)
        {
            _name = name;
            _amount = amount;
            _period = period;
            _festivalrate = festivalrate;
        }

        public string Name
        {
            get
            {
                return _name;
            }
        }

        public double Amount
        {
            get
            {
                return _amount;
            }
        }

        public int Period
        {
            get
            {
                return _period;
            }
        }

        public double CalculateInterest()
        {
            return Amount * Period * _festivalrate.Rate() / 100;
        }

    }
}
